Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -315,7 +311,10 @@ void IOWindow::CreateMainLayout()
     m_option_list->setColumnWidth(1, 64);
     m_option_list->horizontalHeader()->setSectionResizeMode(1, QHeaderView::Fixed);
 
-    m_option_list->setItemDelegate(new InputStateDelegate(this));
+    m_option_list->setItemDelegate(new InputStateDelegate(this, 1, [&](int row) {
+      // Clamp off negative values but allow greater than one in the text display.
```

Comment:
I can see that from the code, but..._why_?
I realize you probably just moved this from `InputStateDelegate::paint` but comments that only reflect what the code does without adding any additional value seem pointless to me.

Reply:
You're not wrong, I was wondering that too. Apparently this was written by @jordan-woyak in #8396, do you happen to remember?

Reply:
Input states may return negative values when opposing inputs are activated ("Axis X-" is negative when "Axis X+" is positive)
https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/InputCommon/ControllerInterface/Device.h#L76

This helps input detection determine the difference between full axis and half axis mappings in a generic manner.
Otherwise it can't tell the difference between "Axis X+" and "Full Axis X+".

Normally this is clamped off by ControlExpression: https://github.com/dolphin-emu/dolphin/blob/master/Source/Core/InputCommon/ControlReference/ExpressionParser.cpp#L267

But because `ControlExpression` is bypassed in this display they have to be manually clamped to not confuse users with weird negative values.

Greater than one values are allowed for accelerometer/gyroscope input.

Reply:
So... should I extend that comment to include the reasoning?